packages/fields/src/fields/use-post-fields/index.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useEntityRecords } from '@wordpress/core-data';
+import type { Field } from '@wordpress/dataviews';
+
+/**
+ * Internal dependencies
+ */
+import featuredImageField from '../featured-image';
+import slugField from '../slug';
+import parentField from '../parent';
+import passwordField from '../password';
+import statusField from '../status';
+import commentStatusField from '../comment-status';
+import titleField from '../title';
+import dateField from '../date';
+import authorField from '../author';
+import type { BasePost, BasePostWithEmbeddedAuthor } from '../../types';
+
+type Post = BasePost | BasePostWithEmbeddedAuthor;
+
+interface UsePostFieldsReturn {
+	isLoading: boolean;
+	fields: Field< Post >[];
+}
+
+interface Author {
+	id: number;
+	name: string;
+}
+
+function usePostFields(): UsePostFieldsReturn {
```

Comment:
Instead of keeping the list of fields here, should we transform this to a "registration API" like actions and start opening the API in Gutenberg? Separate thing obviously, but just want to start this discussion.

Reply:
This is actually a relevant question for this PR: `usePostActions` is defined in the [editor package](https://github.com/WordPress/gutenberg/blob/hide/actions-menu-list/packages/editor/src/components/post-actions/actions.js#L13); but I'm operating under the assumption that we want both to live in `wordpress/fields` at some point, that's why I'm moving it here. Is that your thinking as well?

After that, yeah, field registration would be nice. Probably private to Gutenberg for now?

Reply:
yeah `usePostActions` or more precisely for me `getEntityActions` selector lives in the editor package, mostly because creating new stores is not an easy task and also shouldn't be done at all for bundled packages. So temporarily, the store lives in the editor package. I think we should ultimately do the same for fields. 

And when we decide to make things more public (make the fields package non bundled), we should probably move the store to the fields package.

Reply:
> yeah usePostActions or more precisely for me getEntityActions selector lives in the editor package, mostly because creating new stores is not an easy task and also shouldn't be done at all for bundled packages. So temporarily, the store lives in the editor package. I think we should ultimately do the same for fields.

ok, so this means that, for now, we should move `usePostFields` to the `editor` package in preparation for adding extensibility. Later, when fields is public will move both to this package.

Reply:
That's a good plan. Do you think we should also add the `postType` as param here or in a follow up?

Reply:
Moved to `editor` at https://github.com/WordPress/gutenberg/pull/67024/commits/1b76327d7dad2dba9b7c71ba10292e18914502d1

> That's a good plan. Do you think we should also add the postType as param here or in a follow up?

It's best to do it when we add support for other post types.

Reply:
https://github.com/WordPress/gutenberg/pull/67175 to pull the fields from the registry